ET windows build flags

ivayloen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13638
* #13637
* __->__ #13636

Add compiler flags for windows msvc builds

Differential Revision: [D78854132](https://our.internmc.facebook.com/intern/diff/D78854132/)

ivayloen committed:
ET windows build flags

Add compiler flags for windows msvc builds

Differential Revision: [D78854132](https://our.internmc.facebook.com/intern/diff/D78854132/)

[ghstack-poisoned]

JacobSzwejbka commented on runtime/core/portable_type/c10/c10/targets.bzl:
```diff
@@ -46,7 +46,7 @@ def get_sleef_deps():
             ],
         }),
-        "ovr_config//os:windows": [],
+        "ovr_config//os:windows": ["fbsource//third-party/sleef:sleef"],
```
the way this function is structured seems very strange to me.

ivayloen committed:
Update on "ET windows build flags"


Add compiler flags for windows msvc builds

Differential Revision: [D78854132](https://our.internmc.facebook.com/intern/diff/D78854132/)

[ghstack-poisoned]
